app/puppet-code/deploy.go

```diff
@@ -0,0 +1,67 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/puppetlabs/pe-sdk-go/app/puppet-code/api/client/operations"
+	"github.com/puppetlabs/pe-sdk-go/json"
+	"github.com/puppetlabs/pe-sdk-go/log"
+)
+
+// DeployWithErrorDetails will execute a deploy command and add error details to the error text
+func (puppetCode *PuppetCode) DeployWithErrorDetails(args *DeployArgs) (*operations.DeployOK, error) {
```

Comment:
The `cli` part should not import app, but it should not import `api `
Assuming this is the method called from cli, it should return `string, error` instead  `*operations.DeployOK, error`.

Reply:
Instead of `string` we should probably return a slice of type interface `[]interface{}` since `DeployOK` has multiple types of responses  :
```
type DeployOKBodyItems0 struct {

	// The commit SHA of the control repo that Code Manager used to deploy code in that environment.
	DeploySignature string `json:"deploy-signature,omitempty"`

	// The name of the environment queued or deployed.
	Environment string `json:"environment,omitempty"`

	// error
	Error *models.Error `json:"error,omitempty"`

	// Commit SHAs used internally by file sync to identify the code synced to the code staging directory
	FileSync interface{} `json:"file-sync,omitempty"`

	// Identifies the queue order of the code deploy request.
	ID int64 `json:"id,omitempty"`

	// The status of the code deployment for that environment.
	Status string `json:"status,omitempty"`
}
```